### PR TITLE
[Merged by Bors] - feat(Algebra/Algebra/Basic): prove `algebraMap_eq_zero_iff` for `NoZeroSMulDivisors`

### PR DESCRIPTION
**Pull request by Command-Master**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Can you fix the downstream failures, and add the `eq_one_iff` version too?

**Comment by eric-wieser** [comment]
bors d+

File: Mathlib/FieldTheory/RatFunc/Basic.lean [hunk 1]
```diff
@@ -638,7 +638,6 @@ theorem algebraMap_injective : Function.Injective (algebraMap K[X] (RatFunc K))
   rw [← ofFractionRing_comp_algebraMap]
   exact ofFractionRing_injective.comp (IsFractionRing.injective _ _)
 
-@[simp]
 theorem algebraMap_eq_zero_iff {x : K[X]} : algebraMap K[X] (RatFunc K) x = 0 ↔ x = 0 :=
```
Comment (urkud) [inline Mathlib/FieldTheory/RatFunc/Basic.lean:641]: Can this lemma be deprecated in favor of your new lemma?

**Pull request closed without merging**